src/sinol_make/util.py

```diff
@@ -244,14 +244,28 @@ def stringify_keys(d):
         return d
 
 
-def change_stack_size_to_unlimited():
+def change_stack_size():
     """
-    Function to change the stack size to unlimited.
+    Function to change the stack size to max memory limit.
     """
+    max_memory = 60000 * 1024  # Default memory limit is 66000kB, `resource` module uses bytes.
+    with open(os.path.join(os.getcwd(), "config.yml"), "r") as config_file:
+        config = yaml.load(config_file, Loader=yaml.FullLoader)
+
+    max_memory = max(config.get("memory_limit", 0) * 1024, max_memory)
+    for memory_limit in config.get("memory_limits", {}).values():
+        max_memory = max(memory_limit * 1024, max_memory)
+    for override in config.get("override_limits", {}).values():
+        max_memory = max(override.get("memory_limit", 0) * 1024, max_memory)
+        for memory_limit in override.get("memory_limits", {}).values():
+            max_memory = max(memory_limit * 1024, max_memory)
+    hard_limit = resource.getrlimit(resource.RLIMIT_STACK)[1]
```

Comment:
Any ideas how to shorten the code, or split the functionality into different sections?

Also, I'm not so sure that we care about the "max memory out of all possible options". The `max` function seems not necessarily correct. What do you think?

Reply:
Then what should the stack size be? I thought we decided that it should be max memory limit for the task

Reply:
Even when they would use more memory than they should have? I don't think it's a good solution.

Reply:
Reminder: we want the tool to catch any programmer's mistakes. It's better for a solution to not work *now* than on sio2/during a competition.

Reply:
Right, so why do you think there is a need to set a bigger stack size than the memory limit?

Reply:
Because from what I understand changing stack size is for a whole process, so if the stack size was changed for each solutions while they're running in parallel then it would be changing constantly and some solutions may have wrong stack sizes

Reply:
And from my testing it seems to be the case

Reply:
Sounds good, didn't know about it. Please leave a comment explaining that somewhere.

Reply:
I changed the code a bit. I also deleted the default value since we require that there is a limit for every test